packages/dropcaps/init.lua

```diff
@@ -31,32 +43,68 @@ function package:registerCommands ()
     local scale = SU.cast("number", options.scale or 1.0)
+    local strict = SU.boolean(options.strict, true)
+    local depthadjust = SU.boolean(options.depthadjust, false)
     local color = options.color
     options.size = nil -- we need to measure the "would have been" size before using this
 
     if color then self:loadPackage("color") end
 
+    -- Some initial capital fonts have all their glyphs hanging below the baseline (e.g. EB Garamond Initials)
+    -- We cannot manage all pathological cases.
+    -- Quite empirically, we can shape an "X", which shouldn't usually have a depth normally.
+    -- If it has, then likely all glyphs do also and we need to compensate for that everywhere.
+    local depthAdjustment = depthadjust and shapeHbox(options, { "X" }).depth:tonumber() or 0
+    SU.debug("dropcaps", "Depth adjustment", depthAdjustment)
+
     -- We want the drop cap to span over N lines, that is N - 1 baselineskip + the height of the first line.
     -- Note this only works for the default linespace mechanism.
     -- We determine the height of the first line by measuring the size the initial content *would have* been.
-    -- This gives the font some control over its relative size, sometimes desired sometimes undesired.
     local tmpHbox = shapeHbox(options, content)
     local extraHeight = SILE.measurement((lines - 1).."bs"):tonumber()
-    local targetHeight = tmpHbox.height:tonumber() * scale + extraHeight
+    local curHeight = tmpHbox.height:tonumber() + depthAdjustment
+    if strict then
+      -- Take into account the compensated depth of the initial
+      curHeight = curHeight + tmpHbox.depth:tonumber() - depthAdjustment
+    end
+    local targetHeight = curHeight * scale + extraHeight
     SU.debug("dropcaps", "Target height", targetHeight)
 
     -- Now we need to figure out how to scale the dropcap font to get an initial of targetHeight.
     -- From that we can also figure out the width it will be at that height.
     local curSize = SILE.measurement(SILE.settings:get("font.size")):tonumber()
-    local curHeight, curWidth = tmpHbox.height:tonumber(), tmpHbox.width:tonumber()
+    local curWidth = tmpHbox.width:tonumber()
     options.size = size and size:tonumber() or (targetHeight / curHeight * curSize)
     local targetWidth = curWidth / curSize * options.size
     SU.debug("dropcaps", "Target font size", options.size)
     SU.debug("dropcaps", "Target width", targetWidth)
 
-    -- Typeset the dropcap with its final shape, but don't output it yet
+    -- Typeset the dropcap with its final shape, but don't output it yet.
     local hbox = shapeHbox(options, content)
 
+    -- Compensation for regular extra depth.
+    local compensationHeight = depthAdjustment * options.size / curSize
+
+    if not strict then
+      -- Some fonts have descenders on letters such as Q, J, etc.
+      -- In that case we may need extra lines to the dropcap.
+      local hasExtraDepth = hbox.depth:tonumber() > compensationHeight
```

Comment:
This condition is wrong. I need to check here.

Reply:
Fixed. I have no idea what passed through my mind, trying to compensate for something that was already taken into account...